### PR TITLE
feat: refactor Bookmark component; update actions to addon-actions (#375)

### PR DESCRIPTION
**Pull request by shawncothran**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -56,6 +56,7 @@ export { default as ImageHero } from "./components/imageHero";
 export { default as ItalicText } from "./components/italicText";
 export { default as LastUpdated } from "./components/lastUpdated";
 export { default as Lede } from "./components/lede";
+export { default as ListButton } from "./components/listButton";
```
Comment (thomasthesecond) [inline src/index.js]: Can we just overwrite the bookmark component? Or is this component something different?

File: src/components/listButton/index.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import React, { PropTypes } from "react";
+import radium from "radium";
+import { color } from "../../../settings.json";
```
Comment (thomasthesecond) [inline src/components/listButton/index.jsx]: `import colors from "../../styles/colors";`

File: src/components/listButton/index.jsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import Icon from "../icon";
+
+const styles = {
+  backgroundColor: "rgb(255, 255, 255)",
```
Comment (thomasthesecond) [inline src/components/listButton/index.jsx]: `colors.bgPrimary`

File: src/components/listButton/index.jsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
+import React, { PropTypes } from "react";
```
Comment (thomasthesecond) [inline src/components/listButton/index.jsx]: Replace `{ PropTypes }` with `import PropTypes from "prop-types";`

React throws a warning about this

File: src/components/listButton/index.jsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import radium from "radium";
+import { color } from "../../../settings.json";
+import Icon from "../icon";
+
+const styles = {
+  backgroundColor: "rgb(255, 255, 255)",
+  border: 0,
+  borderRadius: "50%",
+  boxShadow: "rgba(0, 0, 0, 0.1) 0px 0.222222em 0.888889em",
```
Comment (thomasthesecond) [inline src/components/listButton/index.jsx]: `0px` becomes `0`

Can we change the `em` values to show the math? Replace `value` and `fontSize` of course with the actual values.

`rgba(0, 0, 0, 0.1) 0 ${value / fontSize}em ${value / fontSize}em`

File: src/components/listButton/index.jsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  color: color.detailHeaderSmall,
```
Comment (thomasthesecond) [inline src/components/listButton/index.jsx]: I think this color would be `colors.textSecondary`

File: src/components/listButton/index.jsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  cursor: "pointer",
+  display: "inline-block",
+  fontSize: "26px",
```
Comment (thomasthesecond) [inline src/components/listButton/index.jsx]: Let's change this to `fontSizeHeading5` even. This is 24px but we don't have 26px in our style guide.

You'll need to `import { fontSizeHeading5 } from "../../styles/typography";` at the top

File: src/components/listButton/index.jsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  height: "2em",
+  lineHeight: 1,
+  padding: 0,
+  transition: "color 400ms, box-shadow 400ms",
+  width: "2em",
+
+  ":focus": {
+    outline: "1px lightgray dotted",
+    outlineOffset: "2px",
+  },
```
Comment (thomasthesecond) [inline src/components/listButton/index.jsx]: Use the outline mixin here:

`import { outline } from "../../utils/mixins";`

`":focus": outline(),`

File: src/components/listButton/index.jsx [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    position: "relative",
+    transform: "translateX(-50%)",
+  },
+};
+
+const ListButton = ({
+  onClick,
+  size,
+  marked,
+  icon = "BookmarkOutline",
```
Comment (thomasthesecond) [inline src/components/listButton/index.jsx]: This should be set like
```
ListButton.defaultProps = {
  icon: "BookmarkOutline",
};
```

Comment (thomasthesecond) [inline src/components/listButton/index.jsx]: below `propTypes` 👇

File: src/components/listButton/index.jsx [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+  borderRadius: "50%",
+  boxShadow: "rgba(0, 0, 0, 0.1) 0px 0.222222em 0.888889em",
+  color: color.detailHeaderSmall,
+  cursor: "pointer",
+  display: "inline-block",
+  fontSize: "26px",
+  height: "2em",
+  lineHeight: 1,
+  padding: 0,
+  transition: "color 400ms, box-shadow 400ms",
```
Comment (thomasthesecond) [inline src/components/listButton/index.jsx]: `import timing from "../../styles/timing";`

```
transition: `color ${timing.default}`,
```

Not sure we want to transition box-shadow, I think causes CPU usage to spike, could be wrong. Also might not make a noticeable difference here.

File: src/components/listButton/index.jsx [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+  width: "2em",
```
Comment (thomasthesecond) [inline src/components/listButton/index.jsx]: can we also show the math for `width` here and `height` above?

```
width: `${(widthValue / fontSize)}em`,
```

File: src/components/listButton/index.jsx [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+  ":focus": {
+    outline: "1px lightgray dotted",
+    outlineOffset: "2px",
+  },
+};
+
+const iconProps = {
+  label: "Bookmark",
+  style: {
+    color: "#3b434f",
```
Comment (thomasthesecond) [inline src/components/listButton/index.jsx]: `colors.textPrimary`

File: src/components/listButton/index.jsx [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+    display: "block",
+    left: "50%",
+    position: "relative",
+    transform: "translateX(-50%)",
```
Comment (thomasthesecond) [inline src/components/listButton/index.jsx]: Would it be possible to use flexbox here to center vertically and horizontally?

File: src/components/listButton/index.jsx [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+  style,
+}) => {
+  const IconType = Icon[marked ? markedIcon : icon];
+
+  return (
+    <button
+      className="ListButton"
+      style={[
+        styles,
+        size === "large" && { fontSize: "32px" },
```
Comment (thomasthesecond) [inline src/components/listButton/index.jsx]:
```
fontSize: `{fontSizeHeading4}px`,
```

fontSizeHeading4 is 32

File: src/index.js (hunk 1 above)
Comment (shawncothran) [inline src/index.js]: Bookmark is used in a couple places where it makes sense. This component has a few possible uses outside of just as a bookmark icon, where it is an ellipsis, or other icon, that triggers a menu or a modal, etc.

**Comment by shawncothran** [comment]
I did an interactive rebase which ruined the review. Next time I will just push a new commit.

**Pull request merged**